com.unity.hlod/Runtime/Serializer/UserDataSerializerBase.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using UnityEngine;
+
+namespace Unity.HLODSystem.Serializer
+{
+    public abstract class UserDataSerializerBase : MonoBehaviour , ISerializationCallbackReceiver
+    {
+        private Dictionary<int, int> m_idTable = new Dictionary<int, int>();
+
+        [SerializeField]
+        [HideInInspector]
+        private List<int> m_idList = new List<int>();
+        [SerializeField]
+        [HideInInspector]
+        private List<HLODUserData> m_userDataList = new List<HLODUserData>();
+        
+        public void SerializeUserData(int id, GameObject gameObject)
+        {
+            HLODUserData userData = new HLODUserData();
+            SerializeUserData(gameObject, userData);
```

Comment:
This is more like a suggestion which is adding a branch to filter out and not insert the resulting user data into containers when the user data isn't written at all and stays at empty state.
By doing so, we can expect a certain amount of memory to be back!

Reply:
Thanks. Fixed. 

Also it had problem that it was not accumulate data.